controller/behavior/denormalizeable.php

```diff
@@ -0,0 +1,188 @@
+<?php
+/**
+ * @package     LOGman
+ * @copyright   Copyright (C) 2011 - 2016 Timble CVBA. (http://www.timble.net)
+ * @license     GNU GPLv3 <http://www.gnu.org/licenses/gpl.html>
+ * @link        http://www.joomlatools.com
+ */
+
+/**
+ * Denormalizeable Controller Behavior
+ *
+ * Allows for de-normalization of activities entries
+ *
+ * @author  Arunas Mazeika <https://github.com/amazeika>
+ * @package Joomlatools\Component\LOGman
+ */
+class ComActivitiesControllerBehaviorDenormalizeable extends KControllerBehaviorAbstract
+{
+    /**
+     * The resources controller
+     *
+     * @var KModelInterface
+     */
+    protected $_controller;
+
+    public function __construct(KObjectConfig $config)
+    {
+        parent::__construct($config);
+
+        $this->_controller = $config->controller;
+    }
+
+    protected function _initialize(KObjectConfig $config)
+    {
+        $config->append(array(
+            'controller' => 'resource'
+        ));
+
+        if ($this->getIdentifier()->getPackage() != 'activities')
+        {
+            $aliases = array(
+                'com:activities.controller.resource' => array(
+                    'path' => array('controller'),
+                    'name' => 'resource'
+                )
+            );
+
+            foreach ($aliases as $identifier => $alias)
+            {
+                $alias = array_merge($this->getMixer()->getIdentifier()->toArray(), $alias);
+
+                $manager = $this->getObject('manager');
+
+                // Register the alias if a class for it cannot be found.
+                if (!$manager->getClass($alias, false)) {
+                    $manager->registerAlias($identifier, $alias);
+                }
+            }
+        }
+
+        parent::_initialize($config);
+    }
+
+    protected function _afterAdd(KControllerContextInterface $context)
+    {
+        $entity = $context->result;
+
+        if ($entity->action != 'delete')
+        {
+            $context = $this->getMixer()->getContext();
+
+            $context->entity = $entity;
+
+            $this->denormalize($context);
```

Comment:
As discussed on chat:

1. Rename the behavior to ComActivitiesControllerBehaviorResourcable
2. Only store one row per resource, update the row if the resource is updated to reflect it's last state. This make a resource mutable. 
3. Add a uuid to the resource if the resource doesn't have any, if it has one set use that instead (do not add the uuid to the activities table)

I would also hardcode the resource controller and resources table, the activities component is a special system wide component that runs on it own, not as part of another component. In this context we don't want/need  the resource controller to be flexible/configurable. 

Note: this goes back to my initial idea that versioning shouldn't be part of activities. versions should be stored per component/entity, and not globally, which means we need a different component for them. That's fine.